nautobot/extras/models/metadata.py

```diff
@@ -203,16 +223,121 @@ class Meta:
+                #     or self.metadata_type.validation_minimum is not None
+                #     or self.metadata_type.validation_maximum is not None
+                # ):
+                #     json_value = json.dumps(value)
+                #     if (
+                #         self.metadata_type.validation_minimum is not None
+                #         and len(json_value) < self.metadata_type.validation_minimum
+                #     ):
+                #         raise ValidationError(
+                #             f"Value must be at least {self.metadata_type.validation_minimum} characters in length"
+                #         )
+                #     if (
+                #         self.metadata_type.validation_maximum is not None
+                #         and len(json_value) > self.metadata_type.validation_maximum
+                #     ):
+                #         raise ValidationError(
+                #             f"Value must not exceed {self.metadata_type.validation_maximum} characters in length"
+                #         )
+                #     if self.metadata_type.validation_regex and not re.search(
+                #         self.metadata_type.validation_regex, json_value
+                #     ):
+                #         raise ValidationError(f"Value must match regex '{self.metadata_type.validation_regex}'")
+
+            # Validate integer
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_INTEGER:
+                try:
+                    value = int(value)
+                except ValueError:
+                    raise ValidationError("Value must be an integer.")
+                # TODO uncomment this when MetaDataType validation_minimum, validation_maximum and validation_regex fields are implemented
+                # if self.metadata_type.validation_minimum is not None and value < self.metadata_type.validation_minimum:
+                #     raise ValidationError(f"Value must be at least {self.metadata_type.validation_minimum}")
+                # if self.metadata_type.validation_maximum is not None and value > self.metadata_type.validation_maximum:
+                #     raise ValidationError(f"Value must not exceed {self.metadata_type.validation_maximum}")
+
+            # Validate boolean
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_BOOLEAN:
+                try:
+                    value = is_truthy(value)
+                except ValueError as exc:
+                    raise ValidationError("Value must be true or false.") from exc
+
+            # Validate date
+            elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_DATE:
+                if not isinstance(value, date):
+                    try:
+                        datetime.strptime(value, "%Y-%m-%d")
+                    except ValueError:
+                        raise ValidationError("Date values must be in the format YYYY-MM-DD.")
```

Comment:
Also need to implement TYPE_DATETIME (which may be tricky, given timezone support) and TYPE_FLOAT since those aren't yet supported in CustomFields but are valid MetadataTypes.